tests/kernels/quantization/test_nvfp4_moe.py

```diff
@@ -0,0 +1,192 @@
+# SPDX-License-Identifier: Apache-2.0
+import pytest
+import torch
+
+from tests.kernels.utils import torch_moe
+from vllm import _custom_ops as ops
+from vllm.config import ParallelConfig, VllmConfig, set_current_vllm_config
+from vllm.model_executor.layers.fused_moe.cutlass_moe import cutlass_moe_fp4
+from vllm.model_executor.layers.fused_moe.fused_moe import fused_topk
+from vllm.platforms import current_platform
+from vllm.scalar_type import scalar_types
+
+FLOAT4_E2M1_MAX = scalar_types.float4_e2m1fn.max()
+FLOAT8_E4M3_MAX = torch.finfo(torch.float8_e4m3fn).max
+if not current_platform.has_device_capability(100):
+    pytest.skip(reason="Nvfp4 Requires compute capability of 10 or above.",
+                allow_module_level=True)
+
+
+def swizzle_blockscale(scale: torch.tensor):
+    # Pad and blockwise interleave weight_scale
+    assert scale.ndim == 3
+    E, M, K = scale.shape
+    round_up_multiple = lambda x, m: (x + m - 1) // m * m
+    M_padded = round_up_multiple(M, 128)
+    K_padded = round_up_multiple(K, 4)
+    padded_scale = torch.zeros((E, M_padded, K_padded), dtype=scale.dtype)
+    padded_scale[:E, :M, :K] = scale
+    experts, rows, cols = padded_scale.shape
+    assert rows % 128 == 0
+    assert cols % 4 == 0
+    padded_scale = padded_scale.reshape(experts, rows // 128, 4, 32, cols // 4,
+                                        4)
+    swizzled_scale = padded_scale.permute((0, 1, 4, 3, 2, 5))
+    swizzled_scale = swizzled_scale.contiguous().cuda()
+    return swizzled_scale.reshape(E, M, K)
+
+
+kE2M1ToFloat = torch.tensor([0., 0.5, 1., 1.5, 2., 3., 4., 6.],
+                            dtype=torch.float32)
+
+
+def break_fp4_bytes(a, dtype):
```

Comment:
Could you move `break_fp4_bytes`, `convert_swizzled_to_linear`and `dequantize_to_dtype` to a utils file? It would be nice to reuse them across this file and `tests/kernels/quantization/test_nvfp4_scaled_mm.py`

(I also appreciate the nice comments in the other file --> another reason to consolidate)